packages/ymax-planner/src/axelar/gmp-status.ts

```diff
@@ -0,0 +1,346 @@
+      transactionIndex: number;
+    }[];
+  };
+  sourceTransactionHash: string;
+  _id: string;
+  id: string;
+  event: string;
+  transaction: TxMeta & {
+    chainId: number;
+    maxPriorityFeePerGas: string;
+    maxFeePerGas: string;
+    nonce: number;
+  };
+};
+
+type AxelarCallbackEvent = {
+  blockHash: string;
+  chain: string;
+  chain_type: string;
+  address: string;
+  logIndex: number;
+  topics: string[];
+  eventSignature: string;
+  created_at: TimeIndex;
+  transactionIndex: number;
+  eventIndex: number;
+  contract_address: string;
+  transactionHash: string;
+  returnValues: {
+    sender: string;
+    destinationContractAddress: string;
+    payload: string;
+    payloadHash: string;
+    destinationChain: string;
+  };
+  blockNumber: number;
+  block_timestamp: number;
+  receipt: TxReceipt;
+  id: string;
+  event: string;
+  destination_chain_type: string;
+  transaction: TxMeta & {
+    chainId: number;
+    maxPriorityFeePerGas: string;
+    maxFeePerGas: string;
+    nonce: number;
+  };
+  _logIndex: number;
+};
+
+type AxelarEventRecord = {
+  call: AxelarCallEvent;
+  gas_paid: AxelarGasPaidEvent;
+  confirm: AxelarConfirmEvent;
+  approved: AxelarApprovedEvent;
+  executed: AxelarExecutedEvent;
+  callback: AxelarCallbackEvent;
+  // other useful fields
+  message_id: string;
+  status: string;
+  simplified_status: string;
+  is_invalid_call: boolean;
+  is_not_enough_gas: boolean;
+  no_gas_remain: boolean;
+  command_id: string;
+};
+
+type AxelarEventsResponse = {
+  data: AxelarEventRecord[];
+  total: number;
+  time_spent: number;
+};
+
+type EventType = 'ContractCall' | 'ContractCallWithToken';
+
+type StatusType =
+  | 'called'
+  | 'confirming'
+  | 'express_executed'
+  | 'approving'
+  | 'approved'
+  | 'executing'
+  | 'executed'
+  | 'error'
+  | 'waiting_for_route_message'
+  | 'waiting_for_ibc'
+  | 'insufficient_fee';
+
+type AxelarQueryParams = {
+  txHash?: string;
+  txLogIndex?: number;
+  messageId?: string;
+  event?: EventType;
+  commandId?: string;
+  sourceChain?: string;
+  sourceAddress?: string;
+  destinationChain?: string;
+  contractAddress?: string;
+  asset?: string; // use denom or symbol
+  symbol?: string;
+  status?: StatusType;
+  fromTime?: number; // unixtime
+  toTime?: number; // unixtime
+  from?: number; // default: 0
+  size?: number; // default: 25
+};
+
+const MILLIS_PER_MINUTE = 60 * 1000;
+
+// Helpful for experimenting with different parameters:
+// Visit https://docs.axelarscan.io/axelarscan
+export const getTxStatus = async ({
+  url,
+  fetch = globalThis.fetch,
+  params,
+  subscriptionId,
+  logPrefix = '',
+  timeoutMinutes = 5,
+  waitingPeriod = 20,
+}: {
+  url: string;
+  fetch: typeof globalThis.fetch;
+  params: AxelarQueryParams;
+  subscriptionId: string;
+  logPrefix?: string;
+  timeoutMinutes?: number;
+  waitingPeriod?: number;
+}) => {
+  const body = JSON.stringify(params);
+  trace(`${logPrefix} params: ${body}`);
```

Comment:
These `trace` functions returned from `makeTracer` write to standard output, which is not appropriate for library code because stdout should be reserved for consumable data. Suggestion: avoid use of ambient `console` by accepting a `log` power.
```suggestion
  url,
  fetch = globalThis.fetch,
  params,
  subscriptionId,
  log = () => {},
  timeoutMinutes = 5,
  waitingPeriod = 20,
}: {
  url: string;
  fetch: typeof globalThis.fetch;
  params: AxelarQueryParams;
  subscriptionId: string;
  log: (...args: unknown[]) => void;
  timeoutMinutes?: number;
  waitingPeriod?: number;
}) => {
  const body = JSON.stringify(params);
  log(`params: ${body}`);
```